geolib/models/dstability/internal.py

```diff
@@ -1866,6 +1894,13 @@ def add_default_stage(self, label: str, notes: str, unique_start_id=500) -> int:
         ]
         self.states += [State(Id=str(unique_start_id + 3))]
         self.statecorrelations += [StateCorrelation(Id=str(unique_start_id + 4))]
+        self.soillayers += [SoilLayerCollection(Id=str(unique_start_id + 5))]
```

Comment:
These changes are probably great, but don't belong in this PR, right?

Reply:
I discovered by testing that the order of assignment is important in later verions of pydantic.

The block of;
https://github.com/Deltares/GEOLib/blob/d0b9a74e59111a76cc4bb2b993e00fade9bbf1e7/geolib/models/dstability/internal.py#L1904-L1920

Triggers the validation at https://github.com/Deltares/GEOLib/blob/d0b9a74e59111a76cc4bb2b993e00fade9bbf1e7/geolib/models/dstability/internal.py#L1771-L1772

However, when I use the old order here, the setting of `skip_on_failure` got ignored when using `validate_assignment` and thus triggering an error in tests. By adding the sub items first, and afterwards the verification of foreign-keys, this error doesn't show up.